docs/api/StreamReceiveComplete.md

```diff
@@ -17,14 +17,41 @@ void
 
 # Parameters
 
-**TODO**
+- `Stream`: A handle to a valid stream
+- `BufferLength`: The number of bytes processed by the application
 
 # Remarks
 
 This is an asynchronous API but can run inline if called in a callback.
-The application, without setting `StreamMultiReceiveEnabled`, must ensure that one `StreamReceiveComplete` call corresponds to one `QUIC_STREAM_EVENT_RECEIVE` event.
-Duplicate `StreamReceiveComplete` calls after one `QUIC_STREAM_EVENT_RECEIVE` event are ignored silently even with different `BufferLength`.
-The `StreamMultiReceiveEnabled` mode doesn't follow this rule. Multiple `QUIC_STREAM_EVENT_RECEIVE` events can be indicated at once by `StreamReceiveComplete`. The application needs to keep track of accumulated `TotalBufferLength` with this mode.
+
+## Default behavior
+
+The application must ensure that for each `QUIC_STREAM_EVENT_RECEIVE` processed asynchronously (`QUIC_STATUS_PENDING`
+was returned from the callback), `StreamReceiveComplete` is called exactly once.
+
+The application must ensure it calls `StreamReceiveComplete` a single time for every `QUIC_STREAM_EVENT_RECEIVE`
+processed asynchronously. Duplicate `StreamReceiveComplete` calls are ignored silently if no `QUIC_STREAM_EVENT_RECEIVE`
+is pending when the call is processed, but they could race with a new `QUIC_STREAM_EVENT_RECEIVE` event and complete it,
+likely breaking the app data processing.
+
+If `BufferLength` is smaller than the number of bytes indicated in the matching `QUIC_STREAM_EVENT_RECEIVE`, MsQuic will
+stop indicating new `QUIC_STREAM_EVENT_RECEIVE` events until a call to [`StreamReceiveSetEnabled`](StreamReceiveSetEnabled.md).
+
+## Multi-receive Mode
+
+If Multi-receive mode has been enabled on the connection, the behavior is different from default behavior detailed above.
+
+In Multi-receive mode, calls to `StreamReceiveComplete` do not need to match `QUIC_STREAM_EVENT_RECEIVE`: there can be
+either more or less calls to `StreamReceiveComplete` than `QUIC_STREAM_EVENT_RECEIVE` events.
```

Comment:
In this instance, "fewer" is more appropriate than "less" because calls to `StreamReceiveComplete` are countable in the comparative with `QUIC_STREAM_EVENT_RECEIVE` events.

Reply:
```suggestion
either more or fewer calls to `StreamReceiveComplete` than `QUIC_STREAM_EVENT_RECEIVE` events.
```